app/src/main/java/com/example/take_my_money/presentation/viewmodel/FavoriteViewModel.kt

```diff
@@ -12,21 +16,27 @@ class FavoriteViewModel(private val useCaseDataSource: UseCaseDataSource) : View
     val loadDataBase: LiveData<List<CoinEntity>> get() = _loadDataBase
 
     fun loadDataBase() {
-        viewModelScope.launch {
-            _loadDataBase.postValue(useCaseDataSource.getAllCoins())
+        try {
+            viewModelScope.launch {
+                _loadDataBase.postValue(useCaseDataSource.getAllCoins())
+            }
+        } catch (e: Exception) {
+            Log.i("TAG", "loadDataBase: ${e.cause}")
         }
     }
 
-    class FavoriteViewModelFactory(
-        private val repositoryDataSource: RepositoryDataSource
-    ) : ViewModelProvider.Factory {
-
-        override fun <T : ViewModel> create(modelClass: Class<T>): T {
-            return if (modelClass.isAssignableFrom(FavoriteViewModel::class.java)) {
-                FavoriteViewModel(this.repositoryDataSource) as T
-            } else {
-                throw IllegalArgumentException("ViewModel not found")
-            }
-        }
+    fun castListCoinFavorite(coinFavorite: CoinEntity): CoinDomainEntities {
```

Comment:
Change method name, because there's no need for th is List
the method receive just an object/item